infrastructure/src/main/java/org/wso2/testgrid/infrastructure/providers/ShellScriptProvider.java

```diff
@@ -61,15 +61,16 @@ public Deployment createInfrastructure(Infrastructure infrastructure, String inf
         String testPlanLocation = Paths.get(infraRepoDir, "DeploymentPatterns" ,
                 infrastructure.getName()).toString();
 
-        logger.info("Creating the Kubernetes cluster...");
+        logger.info("Executing provisioning scripts...");
         try {
-            TestGridUtil.executeCommand("bash " + Paths.get(testPlanLocation,
-                    getScriptToExecute(infrastructure, Script.ScriptType.INFRA_CREATE)), null);
+            ShellExecutor executor = new ShellExecutor(null);
```

Comment:
Why is null passed the workingDirectory? Can we have a constructor without any arguments instead?